packages/visual-editor/src/components/Directory.tsx

```diff
@@ -429,6 +463,8 @@ const DirectoryComponent = ({ data, styles, analytics }: DirectoryProps) => {
     streamDocument
   );
 
+  const locale = streamDocument.locale || streamDocument?.meta?.locale;
```

Comment:
I'm fine removing this and using i18n.language back in DirectoryCard. Confirmed that the dm_directoryChildren don't have locale so we need to pass it from the top level document.